test_common/KokkosKernels_TestUtils.hpp

```diff
@@ -105,6 +105,94 @@ namespace Test {
       EXPECT_NEAR_KK(h_v1(i), h_v2(i), tol);
     }
   }
-}
 
+  #if defined(KOKKOS_ENABLE_CUDA)
+  using halfScalarType = typename std::conditional<std::is_same<TestExecSpace, Kokkos::Cuda>::value, KokkosKernels::Experimental::device_fp16_t, KokkosKernels::Experimental::host_fp16_t>::type;
+  #else
+  using halfScalarType = KokkosKernels::Experimental::host_fp16_t;
+  #endif // KOKKOS_ENABLE_CUDA
+
+  template<class ViewTypeA, class ViewTypeB, class ViewTypeC, class ExecutionSpace>
+  struct SharedVanillaGEMM {
```

Comment:
Why is this Functor in TestUtils?

Reply:
in an effort to reduce maintenance by providing a single reference GEMM implementation for other test authors to use. should this be moved outside of TestUtils?